### ISSUE TITLE
$ionicPlatform.ready callbacks are ran only once

### ISSUE THREAD

**Issue by dmytro-shchurov** [driftyco/ionic#2380]
Here is my code which let me use `deviceready()` service with plugin wrappers

```
function DeviceReadyPluginWrapperFactory($ionicPlatform) {
        return function (fn) {
            var queue = [];

            var impl = function () {
                queue.push(Array.prototype.slice.call(arguments));
            };

            $ionicPlatform.ready(function () {
                queue.forEach(function (args) {
                    fn.apply(this, args);
                });
                impl = fn;
            }, false);

            return function () {
                return impl.apply(this, arguments);
            };
        };
    }
```

Then, in a wrapper, I call this as

```
function NetworkInformationPluginWrapperFactory($window, deviceReady) {
        var plugin = function () {
            return $window.navigator && $window.navigator.connection;
        };

        return {
            isOnline: deviceReady(function () {
                if (plugin()) {
                    var connectionType = $window.navigator.connection.type;
                    return connectionType !== Connection.NONE;
                } else {
                    return !$window.navigator || $window.navigator.onLine;
                }
            })
        };
    }
```

Until few beta releases ago everything worked fine, and `deviceready(readyCallback)` worked like a charm each time I called it, in case device was really ready. I meant that after a device was ready, when I call `$ionicPlatform.ready(callback)` the `callback` was triggered immediately.
But not now. It seems that it is called only after the platform receives 'load' event.
Dies anybody think that it's ideologically correct behavior? Must I use a 'flag' and bypass `$ionicPlatform.ready()` after it runs once?
